### ISSUE TITLE
Able to save attributes with null for attribute_code

### ISSUE THREAD

**Issue by will-b** [magento/magento2#8064]
### Preconditions

1. Magento version 2.1.3

### Steps to reproduce

1. Create a setup script,  
2. Add a new EAV attribute with a null for the attribute code
```
$attributeData = [.......];
$eavSetup =  $this->eavSetupFactory->create(['setup' => $setup]);
$eavSetup->addAttribute(Product::ENTITY, null, $attributeData);
```
3. Run setup

### Expected result

1. I expected that the attribute will not be created, and I expect to see and error or to have an error logged.
2. I expect the database column `attribute_code` to not allow null values
### Actual result

1. The attributed is created and has null has an attributed code
DDL for eav_attribute.attribute_code. 
```
`attribute_code` varchar(255) DEFAULT NULL COMMENT 'Attribute Code',
```

I would expected the "Default Label" for an attribute to also not allow null values.

**Comment by vherasymenko** [comment]
hello @will-b , Thank for your report, we create internal ticket MAGETWO-67198

**Comment by magento-engcom-team** [comment]
@will-b, thank you for your report.
The issue is already fixed in 2.2.0
